data/integration_tests/percolator/IntegrationTest_Percolator_Speed.py

```diff
@@ -0,0 +1,242 @@
+url = "https://kth.box.com/shared/static/t3dgd186n3ckfj5qvhpxga4haiz8lmh8.gz"
+
+def checkIfTestDataExist():
+    path = Path(testDataFolderPath)
+    path.mkdir(parents=False, exist_ok=True)
+    fileLocation = Path(testDataFullPath)
+    return fileLocation.is_file()
+
+def downloadTestData():
+    req = Request(url, headers={'User-Agent': 'XYZ/3.0'})
+    site = urlopen(req, timeout=20)
+    meta = site.info()
+    file_size = int(meta.get_all("Content-Length")[0])
+    print ("Downloading: %s Bytes: %s" % (downloadedDataFileName, file_size))
+    f = open(testDataFolderPath + "/" + downloadedDataFileName, 'wb')
+    file_size_dl = 0
+    block_sz = 8192
+    prevDownloadProgress = 0
+    while True:
+        buffer = site.read(block_sz)
+        if not buffer:
+            break
+        file_size_dl += len(buffer)
+        f.write(buffer)
+        downloadProgress = file_size_dl * 100. / file_size
+        status = r"%10d  [%.0f%%]" % (file_size_dl, downloadProgress)
+        status = status + chr(8)*(len(status)+1)
+        if (downloadProgress - prevDownloadProgress > 1. or downloadProgress > 99.9):
+            print (status, end="")
+            prevDownloadProgress = downloadProgress
+    f.close()
+    print("\nDownload complete.\n")
+
+def extractTarFile(fname):
+    if fname.endswith("tar.gz"):
+        tar = tarfile.open(fname, "r:gz")
+        tar.extractall(testDataFolderPath)
+        tar.close()
+    elif fname.endswith("tar"):
+        tar = tarfile.open(fname, "r:")
+        tar.extractall(testDataFolderPath)
+        tar.close()
+
+def getArguments():
+    parser = ArgumentParser(description="Measure speed of perculator application.")
+    parser._action_groups.pop()
+    required = parser.add_argument_group('required arguments')
+    optional = parser.add_argument_group('optional arguments')
+
+    required.add_argument('-d','--data', type=str, default = "" , metavar='', required=False, help="Path to input-data used by percolator.")
+    required.add_argument('-r','--runs', type=int, metavar='', default=3, required=False, help="Number of tests to evaluate.")
+    optional.add_argument('-f','--flags', type=str, default = "" , metavar='', required=False, help="Flags to use with the executable.")
+    optional.add_argument('-a','--await_tests', default=False, action="store_true", required=False, help="Wait for others processes running this script to finish before starting.")
+    optional.add_argument('-p','--processes', nargs='+', type=int, metavar='',required=False, help="Other test-PIDs to wait for before starting test(s).")
+    optional.add_argument('-c','--comments', type=str, default="", metavar='', required=False, help="Comments regarding test details.")
+    return parser.parse_args()
+
+def awaitOtherTasks(args):
+    if(args.processes is not None):
+        waitForProcesses(args.processes)
+    if(args.await_tests is True):
+        waitForOtherTests()
+
+def waitForProcess(pid):
```

Comment:
psutil is already used elsewhere so might as well use it consistently.

Reply:
That is a good argument for using it here, yes. My only reservation is that psutil is not a standard library, so it might be better to minimize dependency on it? I feel like psutil shouldn't actually be necessary for anything in this code, but this may be my Unix bias showing. In any case, it's a minor point, so I don't feel the need to pursue it further.